test/Infrastructure/LeanCode.Kratos.Tests/KratosWebHookHandlerTests.cs

```diff
@@ -0,0 +1,173 @@
+        var ctx = await RunAsync(handler, new() { [handler.ApiKeyHeaderName] = Guid.NewGuid().ToString() });
+
+        Assert.Equal(403, ctx.Response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Runs_inner_handler_if_api_key_is_valid()
+    {
+        var handler = PrepareHandler(ctx => ctx.Response.StatusCode = 200);
+        var ctx = await RunAsync(handler, new() { [handler.ApiKeyHeaderName] = ApiKey });
+
+        Assert.Equal(200, ctx.Response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Swallows_unhandled_exceptions_thrown_by_inner_handler_and_responds_with_status_code_500()
+    {
+        var handler = PrepareHandler(ctx => throw new InvalidOperationException());
+        var ctx = await RunAsync(handler, new() { [handler.ApiKeyHeaderName] = ApiKey });
+
+        Assert.Equal(500, ctx.Response.StatusCode);
+    }
+
+    [Fact]
+    public async Task Correctly_serializes_identity_responses()
+    {
+        var email = "test@leancode.pl";
+        var timestampString = "2023-06-26T21:37:42Z";
+        var timestamp = DateTime.Parse(timestampString, CultureInfo.InvariantCulture).ToUniversalTime();
+
+        var identity = new Identity
+        {
+            Id = Guid.Parse("00000000-0000-0000-0000-000000000001"),
+            CreatedAt = timestamp,
+            UpdatedAt = timestamp,
+            SchemaId = "user",
+            SchemaUrl = new("https://auth.local.lncd.pl/schemas/dXNlcg"),
+            State = IdentityState.Active,
+            RecoveryAddresses = new(1)
+            {
+                new()
+                {
+                    Id = Guid.Parse("00000000-0000-0000-0000-000000000002"),
+                    CreatedAt = timestamp,
+                    UpdatedAt = timestamp,
+                    Via = AddressType.Email,
+                    Value = email,
+                },
+            },
+            VerifiableAddresses = new(1)
+            {
+                new()
+                {
+                    Id = Guid.Parse("00000000-0000-0000-0000-000000000003"),
+                    CreatedAt = timestamp,
+                    UpdatedAt = timestamp,
+                    Via = AddressType.Email,
+                    Value = email,
+                    Status = AddressStatus.Completed,
+                    VerifiedAt = timestamp,
+                    Verified = true,
+                },
+            },
+            Traits = JsonSerializer.SerializeToElement(new Dictionary<string, string> { ["email"] = email }),
+            MetadataPublic = null,
+            MetadataAdmin = null,
+        };
+
+        var handler = PrepareHandler(ctx => KratosWebHookHandlerBase.WriteIdentityResponseAsync(ctx, identity));
+        var ctx = await RunAsync(handler, new() { [handler.ApiKeyHeaderName] = ApiKey });
+
+        Assert.Equal(200, ctx.Response.StatusCode);
+
+        ctx.Response.Body.Position = 0L;
+        using var document = await JsonDocument.ParseAsync(ctx.Response.Body);
```

Comment:
I can deserialize to `Identity`, I don't have to use FluentAsssertions to compare them, but in any case I think it would defeat the purpose because it won't allow me to verify over-the-wire format, if something was broken in the serialization code and e.g. enums were serialized as numbers, Kratos could fail to understand our response and we would be unable to detect it in tests.

Reply:
Ah, got it. Valid point. So, let's leave it as-is indeed.